### PR TITLE
Update party-robot introduction.md

### PR DESCRIPTION
**Pull request by btsui3**
- Issue #1728
- Added information about Golang packages
- Added information about defining public vs private in packages

### CODE REVIEW

File: exercises/concept/party-robot/.docs/introduction.md [hunk 1]
```diff
@@ -1,5 +1,7 @@
 # Introduction
 
+Go programs are organized into packages. A package is a collection of source files in the same directory that are compiled together. Functions, types, variables, and constants defined in one source file are visible to all other source files within the same package. Similar to program languages like Java that use access modifiers such as `public` and `private`, Go determines if an item is exported and unexported through how it is declared. To allow an item to be called by code in other packages, the name must start with a capital letter.
```
Comment (bobtfish) [inline exercises/concept/party-robot/.docs/introduction.md:3]: Can you split this into one line per sentence. A lot of the text isn't like this at the moment, but we're trying to change it as we touch things as this makes diffs much easier to read.

Also this is the exercise's introduction.md - at least some of this content needs adding to the concept's introduction.md document (`concepts/packages/introduction.md`) and also the about.md document (`concepts/packages/about.md`).

**Review by bobtfish — CHANGES_REQUESTED** [review]
Probably also worth providing a code sample to show public vs private. something like:

````
```go
package greeting
// Hello is a public function (callable from other packages)
func Hello(name string) string {
    return "Hello " + name
}
// hello is a private function (not callable from other packages)
func hello(name string) string {
    return "Hello " + name
}
```
````

I'd also link a reference, something like https://golang.org/ref/spec#Exported_identifiers in the `concepts/packages/about.md`

**Comment by bobtfish** [comment]
Thanks for doing this <3

File: exercises/concept/party-robot/.docs/introduction.md (hunk 1 above)
Comment (kahgoh) [inline exercises/concept/party-robot/.docs/introduction.md:3]: > To allow an item to be called by code in other packages, the name must start with a capital letter.

I'd like to suggest adding an example would help to make this clearer.

Comment (btsui3) [reply exercises/concept/party-robot/.docs/introduction.md]: @bobtfish Thanks for the helpful comments. 

I'm new to the `concepts` feature - how do I know what information to put in the `about.md` vs `introduction.md` ?

Comment (bobtfish) [reply exercises/concept/party-robot/.docs/introduction.md]: Hey @btsui3 - there's some guidance here: https://exercism.org/docs/building/tracks/concepts

tl;dr: `introduction.md` should only contain material needed to solve the related exercise, no links - this is what's shown before the student has completed the exercise. `about.md` can contain a superset of this information / a broader picture and will be shown once the exercise is completed, it can / should link out to other documentation.

They are often extremely similar at the moment as we've been beating things into shape in Go, over time they'll probably diverge more!

File: exercises/concept/party-robot/.docs/introduction.md [hunk 2]
```diff
@@ -1,5 +1,25 @@
 # Introduction
 
+Go programs are organized into packages. 
+A package is a collection of source files in the same directory that are compiled together. 
+Functions, types, variables, and constants defined in one source file are visible to all other source files within the same package. 
+Similar to program languages like Java that use access modifiers such as `public` and `private`, Go determines if an item is exported and unexported through how it is declared. 
+To allow an item to be called by code in other packages, the name must start with a capital letter.
```
Comment (bobtfish) [inline exercises/concept/party-robot/.docs/introduction.md:7]: This (line 7+) isn't needed to solve the exercise, and so I don't think we want to include it in the exercise really.

The intro is great here, but the public/private part should, I think, only really be in `about.md`

Comment (btsui3) [reply exercises/concept/party-robot/.docs/introduction.md]: Sounds good - added two commits for moving the public/private info from `party-robot/introduction.md` to `concept/packages/about.md`

File: exercises/concept/party-robot/.docs/introduction.md [hunk 3]
```diff
@@ -1,5 +1,9 @@
 # Introduction
 
+Go programs are organized into packages. 
+A package is a collection of source files in the same directory that are compiled together. 
+Functions, types, variables, and constants defined in one source file are visible to all other source files within the same package.
+
```
Comment (junedev) [inline exercises/concept/party-robot/.docs/introduction.md:6]: This introduction should also include a short section on what is exported from a package.

File: concepts/packages/about.md [hunk 4]
```diff
@@ -52,3 +52,18 @@ errors.New("Connection not established")
...
+package greeting
+// Hello is a public function (callable from other packages)
+func Hello(name string) string {
+    return "Hello " + name
+}
+// hello is a private function (not callable from other packages)
+func hello(name string) string {
+    return "Hello " + name
+}
+```
```
Comment (junedev) [inline concepts/packages/about.md:69]: This content (or a shorter version of it) is probably also missing from the concept introduction.md file.

File: concepts/packages/about.md [hunk 5]
```diff
@@ -52,3 +52,18 @@ errors.New("Connection not established")
 // using the errors package located at github.com/pkg/errors by alias
 errs.New("Connection not established")
 ```
+
+Similar to program languages like Java that use access modifiers such as `public` and `private`, Go determines if an item is exported and unexported through how it is declared.
```
Comment (junedev) [inline concepts/packages/about.md:56]: The term "similar" was a bit confusing for me because in the end its not very similar when you look at the syntax.
```suggestion
Languages like Java use access modifiers such as `public` and `private` to specify what is accessible from outside. Go determines whether an entity is exported from a package or not through how it is declared.
```

**Review by junedev — COMMENTED** [review]
Thanks for taking care of this. 

I left some comments especially regarding making sure the content of the concept introduction and about and the exercise introduction stays in sync.

**Comment by btsui3** [comment]
Sorry about bailing on this PR. I really appreciate everyone helping me, but I think someone with more project experience wil have more success with resolving this issue.

**Comment by junedev** [comment]
No problem, someone else can pick this up. I know the work on these concept exercises can be overwhelming at first. It also took me a while to wrap my head around it. Maybe you can try again later on some other issue.

**Comment by junedev** [comment]
@bobtfish Would you be able to pick this up and fix the remaining comments? (Probably its easiest you make a new PR and copy the content you want to include.)

**Comment by bobtfish** [comment]
Yeah, I can pick this up no problem.

Thanks for doing most of the heavy lifting @btsui3 <3

**Comment by bobtfish** [comment]
New PR is #1765
